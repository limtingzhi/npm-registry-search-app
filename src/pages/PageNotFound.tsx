import { NavLink } from 'react-router';

function PageNotFound() {
  return (
    <div>
      <h1>404</h1>
      <p>We can't seem to find the page you're looking for.</p>
      <NavLink to="/">Go Home</NavLink>
    </div>
  );
}

export default PageNotFound;
