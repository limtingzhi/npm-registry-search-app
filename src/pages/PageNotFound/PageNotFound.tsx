import { Button, Container, Stack, Typography } from '@mui/material';
import { Link } from 'react-router';

function PageNotFound() {
  return (
    <Container sx={{ paddingY: 3 }}>
      <Stack spacing={3} sx={{ alignItems: 'center', textAlign: 'center' }}>
        <Typography gutterBottom variant="h1">
          404
        </Typography>
        <Typography gutterBottom variant="body1">
          We can't seem to find the page you're looking for.
        </Typography>
        <Button component={Link} disableElevation to="/" variant="contained">
          Go Home
        </Button>
      </Stack>
    </Container>
  );
}

export default PageNotFound;
