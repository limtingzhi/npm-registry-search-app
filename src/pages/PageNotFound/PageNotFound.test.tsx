/**
 * @jest-environment jsdom
 */

import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import PageNotFound from './PageNotFound';

describe('PageNotFound', () => {
  test('should match snapshot with default state', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <PageNotFound />
      </MemoryRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
