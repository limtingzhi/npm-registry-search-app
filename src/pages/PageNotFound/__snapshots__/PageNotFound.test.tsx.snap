// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNotFound should match snapshot with default state 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-j81c45-MuiContainer-root"
  >
    <div
      class="MuiStack-root css-a76xbi-MuiStack-root"
    >
      <h1
        class="MuiTypography-root MuiTypography-h1 MuiTypography-gutterBottom css-dyz9rb-MuiTypography-root"
      >
        404
      </h1>
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-vjrxpc-MuiTypography-root"
      >
        We can't seem to find the page you're looking for.
      </p>
      <a
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation css-1cu450-MuiButtonBase-root-MuiButton-root"
        data-discover="true"
        href="/"
        tabindex="0"
      >
        Go Home
      </a>
    </div>
  </div>
</DocumentFragment>
`;
